lib/mix/lib/mix/utils.ex

```diff
@@ -464,16 +464,30 @@ defmodule Mix.Utils do
   defp proxy_env do
     http_proxy  = System.get_env("HTTP_PROXY")  || System.get_env("http_proxy")
     https_proxy = System.get_env("HTTPS_PROXY") || System.get_env("https_proxy")
+    no_proxy    = System.get_env("NO_PROXY")    || System.get_env("no_proxy")
+                |> no_proxy_list()
```

Comment:
Parenthesis are needed around the expression before `|>`. May be better to change to a function call instead and ignore the aligning?